src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java

```diff
@@ -654,6 +654,20 @@ public void testLoadConfigurationFromClassPath() throws Exception {
             .isEqualTo(0);
     }
 
+    @Test
+    public void testLoadConfigurationFromClassPathWithNonAsciiSymbolsInPath() throws Exception {
+        final DefaultConfiguration config =
+                (DefaultConfiguration) ConfigurationLoader.loadConfiguration(
+                    getResourcePath("æ£µ¥/InputConfigurationLoaderModuleIgnoreSeverity.xml"),
+                        new PropertiesExpander(new Properties()), IgnoredModulesOptions.OMIT);
+
+        final Configuration[] children = config.getChildren();
+        final int length = children[0].getChildren().length;
+        assertWithMessage("Invalid children count")
+            .that(length)
+            .isEqualTo(0);
```

Comment:
This very confusing, why  0 is good signal of data loading from file.
Let's check module name, or fact that severity property is ignore.

Reply:
I copied from test right above it. And 5 more tests with same assertion (children count is zero)
So it makes more sense to fix it in separate issue/commit/PR, since it is not relevant to this issue.

Imo it is not confusing, it means that file content is irrelevant, the only thing that matters is the fact that it was loaded without errors.

Reply:
We should not have tests that don't care about content.
Let's break this bad practice of pure unit testing and simply run some validation on some Check and get any violation. In this case we will just be sure, whole checkstyle process works.

Reply:
> Let's break this bad practice of pure unit testing and simply run some validation on some Check and get any violation. In this case we will just be sure, whole checkstyle process works.

Sorry, but that does not make any sense to me. This UT class is for testing configuration loader. **All** its tests test loading of the configuration and just verify presence of some nodes. From this point of view verification of the empty content is also a perfectly valid use case.

What you are talking about is a kind of end-to-end testing, not UT for specific class.

Reply:
>All its tests test loading of the configuration and just verify presence of some nodes. From this point of view verification of the empty content is also a perfectly valid use case.

This two sentences contradict to each other.

If start in partial integration target of test - we should load config and briefly verify that content is loaded, severity=ignore is seen.
Or we just reuse end-to-end model and show that all is working for user.

Reply:
> we just reuse end-to-end model and show that all is working for user

Once again, this does not make sense. This is UT for ConfigurationLoader, which is supposed to test configuration loading, not how whole checkstyle works.

> This two sentences contradict to each other.

They are not. Loading of the empty config can happen, this input is completely valid and accepted by code. So, code should be able to work with empty file and produce no error and empty configuration. Why I should not test the valid input and verify the result? who knows, may be empty config will produce some unexpected stuff in config?

> we should load config and briefly verify that content is loaded

This is exactly what happens here. No checks/filters in file - no nodes in config. The point is to check that loading completes without an exception.
If it makes you feel better, I can replace it with something like

```
        final Properties expectedPropertyValues = new Properties();
        expectedPropertyValues.setProperty("tabWidth", "2");
        expectedPropertyValues.setProperty("basedir", ".");
        expectedPropertyValues.setProperty("charset", "ASCII");
        verifyConfigNode(config, "Checker", 0, expectedPropertyValues);
```

Reply:
I missing a point why not Check in config:
![image](https://github.com/checkstyle/checkstyle/assets/812984/178f605e-21c0-4c82-9679-3d394ae2f29c)

Reply:
> IgnoredModulesOptions.OMIT);

please use different value EXECUTE.
and please check that in config we have MemberName with property.

Reply:
I changed it to verify node